Source/Core/src/ca/uqac/lif/synthia/random/RandomString.java

```diff
@@ -37,41 +35,99 @@
 	 */
 	protected Picker<Integer> m_lengthPicker;
 
-	protected RandomInteger m_charValuePicker;
+	protected RandomInteger m_charIndexPicker;
 
+	protected char[] m_chars;
 
 	/**
-	 * Creates a new RandomString picker
+	 * Creates a new RandomString picker with a default alphanumeric values array
+	 *
 	 * @param length A picker used to determine the string's length
 	 */
 	public RandomString(Picker<Integer> length)
 	{
 		super();
+		defaultCharArrayInitialize();
 		m_lengthPicker = length;
-		m_charValuePicker = new RandomInteger(48,122); //ASCII numeric value range
+		m_charIndexPicker = new RandomInteger(0, m_chars.length - 1);
+	}
+
+	/**
+	 * Creates a new RandomString picker with a default alphanumeric values array
+	 *
+	 * @param length A picker used to determine the string's length
+	 * @param char_array An array containing the characters allowed in the random string.
+	 */
+	public RandomString(Picker<Integer> length, char[] char_array)
```

Comment:
One small suggestion here: replace the second argument from `char[] char_array` to `char ... char_array`. The three dots mean that we can pass a variable number of char arguments, which are passed to the constructor as an array. This allows us to write, for example:

```java
new RandomString(new RandomInteger(), 'a', 'b', 'c')`
```

instead of

```java
new RandomString(new RandomInteger(), new char[]{ 'a', 'b', 'c'})`
```

(This second syntax is still also possible.)